src/main/java/net/sf/jabref/cli/ArgumentProcessor.java

```diff
@@ -0,0 +1,548 @@
+package net.sf.jabref.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.prefs.BackingStoreException;
+
+import net.sf.jabref.BibDatabaseContext;
+import net.sf.jabref.Defaults;
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefException;
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.MetaData;
+import net.sf.jabref.exporter.BibDatabaseWriter;
+import net.sf.jabref.exporter.ExportFormats;
+import net.sf.jabref.exporter.IExportFormat;
+import net.sf.jabref.exporter.SaveException;
+import net.sf.jabref.exporter.SavePreferences;
+import net.sf.jabref.exporter.SaveSession;
+import net.sf.jabref.importer.AutosaveAwareDatabaseLoader;
+import net.sf.jabref.importer.ImportFormatReader;
+import net.sf.jabref.importer.ImportInspectionCommandLine;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fetcher.EntryFetcher;
+import net.sf.jabref.importer.fetcher.EntryFetchers;
+import net.sf.jabref.logic.CustomEntryTypesManager;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.labelpattern.LabelPatternUtil;
+import net.sf.jabref.logic.logging.JabRefLogger;
+import net.sf.jabref.logic.search.DatabaseSearcher;
+import net.sf.jabref.logic.search.SearchQuery;
+import net.sf.jabref.logic.util.OS;
+import net.sf.jabref.logic.util.strings.StringUtil;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.util.Util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ArgumentProcessor {
+
+    public enum Mode {
+        INITIAL_START,
+        REMOTE_START
+    }
+
+
+    private static final Log LOGGER = LogFactory.getLog(ArgumentProcessor.class);
+
+    private final JabRefCLI cli;
+
+    private final Optional<List<ParserResult>> parserResults;
+
+    private final Mode startupMode;
+
+
+    public ArgumentProcessor(String[] args, Mode startupMode) {
+        cli = new JabRefCLI(args);
+        this.startupMode = startupMode;
+        parserResults = processArguments();
+    }
+
+    public Optional<List<ParserResult>> getParserResults() {
```

Comment:
Btw, an empty list can be returned although the call is correct and valid, e.g. everytime JabRef is started without any argument, so some additional logic was required to deal with the erroneous cases (which is why the Optional was there in the first place).